packages/aws-cdk/lib/commands/doctor.ts

```diff
@@ -68,3 +68,10 @@ function displayCdkEnvironmentVariables() {
   }
   return healthy;
 }
+
+function anonymizeAwsVariable(name: string, value: string) {
+  if (name === 'AWS_ACCESS_KEY_ID') { return value.substr(0, 4) + '*'.repeat(Math.max(0, value.length - 4)); }
```

Comment:
And here, I would put `####<redacted>`. I also don't know whether the "common" practice is to show few first or few last... Maybe should research some weak evidence of what others do (we don't want to cause leaking a head when other tools leak the tail - that'd be us giving out additional info)?

Reply:
I'm leaking the first 4 characters on purpose, because they tell us the *type* of access key used:

* AKIA for long-lived (user) keys.
* ASIA for short-lived (session, role) keys.

These are standardized and not part of the secret material.